framework/editor/LazyLines.kt

```diff
@@ -142,7 +142,7 @@ internal object LazyLines {
             .mouseScrollFilter { event, _ -> onScroll(); state.scroller.updateOffset(event) }
             .onSizeChanged { state.scroller.viewHeight = toDP(it.height, density) }) {
             if (state.lines.isNotEmpty()) {
-                val lastVisibleIndex = min(state.scroller.lastVisibleIndexPossible, state.scroller.lineCount() - 1)
+                val lastVisibleIndex = min(state.scroller.lastVisibleIndexPossible + 1, state.scroller.lineCount() - 1)
```

Comment:
In line with our new padding approach that only scrolls when the cursor is below the visible text editor area, we need to generate the last visible index + 1 so that when the cursor goes below the last visible line, it is recognised as having moved onto the next row. If we don't visualise this line, the selection won't extend to the next line and as such won't scroll.